cylc/flow/scripts/stop.py

```diff
@@ -240,27 +240,33 @@ async def _run(
             options.max_polls,
         )
 
-    # mode defaults to 'Clean'
-    mode = None
     if stop_task or stop_cycle:
-        pass
+        mode: Optional[str] = None
     elif options.kill:
         mode = WorkflowStopMode.Kill.name
     elif options.now > 1:
         mode = WorkflowStopMode.NowNow.name
     elif options.now:
         mode = WorkflowStopMode.Now.name
+    else:
+        mode = WorkflowStopMode.Clean.name
+
+    variables = {
+        'wFlows': [workflow_id],
+        'stopMode': mode,
+        'cyclePoint': stop_cycle,
+        'clockTime': options.wall_clock,
+        'task': stop_task,
+    }
+    # Only include argument if set.
+    if options.flow_num is not None:
+        if options.flow_num.isdigit():
+            options.flow_num = int(options.flow_num)
+        variables['flowNum'] = options.flow_num
```

Comment:
Doesn't quite do it for back-compat:
```console
$ cylc stop mwflow
RequestError: Please check your query variables. The following variables are missing: [flowNum]
(Workflow is running in Cylc 8.4.2)
```
Although the workflow does shut down.

Reply:
Well, I can't change the old scheduler end, the new client shouldn't pass in `None`, and `flow_num` doesn't have a default in this case..
Well stop can handle `None` in the backend commands, so I suppose I could reinstate it .. (since it doesn't have a graphql default):
```
@_command('stop')
async def stop(
    schd: 'Scheduler',
    mode: 'Optional[Enum]',
    cycle_point: Optional[str] = None,
    # NOTE clock_time YYYY/MM/DD-HH:mm back-compat removed
    clock_time: Optional[str] = None,
    task: Optional[str] = None,
    flow_num: Optional[int] = None,
):
    if task:
        validate.is_tasks([task])
    yield
    if flow_num:
        schd.pool.stop_flow(flow_num)
        return
```

Reply:
I'm fairly sure the coercion was necessary (but this is only by memory, and that I wouldn't have done this for no reason), and I changed the non-inclusion on None at the same time...
(any string version of an int (which comes in from CLI) would have tripped it up)

Anyway, fixed (hopefully), give it a try

Reply:
Yes, with this commented out:
```
+    #if options.flow_num is not None and options.flow_num.isdigit():
+    #    options.flow_num = int(options.flow_num)
```
![image](https://github.com/user-attachments/assets/8eb67860-eec8-4ce8-85cd-180224d871fe)


So I'm leaving it in

Reply:
Ah right, strangely it is not needed for back-compat in either direction, but only for "now-compat"